src/components/ActionBar.tsx

```diff
@@ -39,30 +41,37 @@ function ActionBar(props: ActionBarProps) {
             );
             console.log(e);
         }
-        setSmartWallets([]);
+        tempSmartWallets.push(...smartWallets);
         for (let i = 0; i < tempSmartWallets.length; i += 1) {
             Utils.getSmartWalletBalance(tempSmartWallets[i], state.token!).then(
-                (tempSmartWallet) =>
-                    setSmartWallets((prev) => [...prev, tempSmartWallet])
+                (tempSmartWallet) => {
+                    const exists = smartWallets.find(
+                        (x) => x.address === tempSmartWallet.address
+                    );
+                    if (exists) {
+                        setSmartWallets((prev) =>
+                            prev.map((x) => {
+                                if (x.address === tempSmartWallet.address) {
+                                    return tempSmartWallet;
+                                }
+                                return x;
+                            })
+                        );
+                    } else {
+                        setSmartWallets((prev) => [...prev, tempSmartWallet]);
+                    }
+                }
             );
         }
     };
 
     useEffect(() => {
+        if (!updateInfo) {
+            return;
+        }
         (async () => {
             if (state.token) {
                 await loadSmartWallets();
-                const workerAddress =
-                    process.env.REACT_APP_CONTRACTS_RELAY_WORKER!;
-                const currentWorkerBalance = parseFloat(
-                    Utils.fromWei(
-                        await Utils.tokenBalance(
-                            workerAddress,
-                            state.token!.address
-                        )
-                    )
-                ).toFixed(4);
-                setWorkerBalance(currentWorkerBalance);
             }
         })();
     }, [state.token, updateInfo]);
```

Comment:
this is bad bad... using async functions in `useEffect` is very much discouraged as it results in race conditions. For such situations use `useCallback`.

Reply:
I forgot to remove the async and await. This is not a async function.

Reply:
updated